openmdao/api.py

```diff
@@ -40,10 +39,8 @@
 from openmdao.solvers.run_once import RunOnce
 from openmdao.solvers.scipy_gmres import ScipyGMRES
 from openmdao.solvers.solver_base import LinearSolver, NonLinearSolver
-try:
+if under_mpirun():
     from openmdao.solvers.petsc_ksp import PetscKSP
```

Comment:
Error obtaining unique transport key from ORTE (orte_precondition_transports not present in
the environment).

  Local host: jess.dtu.dk
--------------------------------------------------------------------------
--------------------------------------------------------------------------
It looks like MPI_INIT failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during MPI_INIT; some of which are due to configuration or environment
problems.  This failure appears to be an internal failure; here's some
additional information (which may only be relevant to an Open MPI
developer):

  PML add procs failed
  --> Returned "Error" (-1) instead of "Success" (0)
--------------------------------------------------------------------------
[jess.dtu.dk:60960] *** An error occurred in MPI_Init_thread
[jess.dtu.dk:60960] *** on a NULL communicator
[jess.dtu.dk:60960] *** Unknown error
[jess.dtu.dk:60960] *** MPI_ERRORS_ARE_FATAL: your MPI job will now abort
--------------------------------------------------------------------------
An MPI process is aborting at a time when it cannot guarantee that all
of its peer processes in the job will be killed properly.  You should
double check that everything has shut down cleanly.

  Reason:     Before MPI_INIT completed
  Local host: jess.dtu.dk
  PID:        60960